apps/meteor/tests/e2e/omnichannel/omnichannel-take-chat.spec.ts

```diff
@@ -25,50 +27,66 @@ test.describe('omnichannel-takeChat', () => {
 			await api.post('/settings/Livechat_enabled_when_agent_idle', { value: false }).then((res) => expect(res.status()).toBe(200)),
-			await api.delete('/livechat/users/agent/user1').then((res) => expect(res.status()).toBe(200)),
 		]);
 
 		await agent.page.close();
 	});
 
-	test.beforeEach(async ({ page, api }) => {
-		// make "user-1" online
+	test.beforeEach('start a new livechat chat', async ({ page, api },) => {
 		await agent.poHomeChannel.sidenav.switchStatus('online');
 
-		// start a new chat for each test
 		newVisitor = {
 			name: `${faker.person.firstName()} ${faker.string.uuid()}`,
 			email: faker.internet.email(),
 		};
+
 		poLiveChat = new OmnichannelLiveChat(page, api);
+
 		await page.goto('/livechat');
-		await poLiveChat.openLiveChat();
-		await poLiveChat.sendMessage(newVisitor, false);
-		await poLiveChat.onlineAgentMessage.type('this_a_test_message_from_user');
-		await poLiveChat.btnSendMessageToOnlineAgent.click();
 	});
 
-	test('expect "user1" to be able to take the chat from the queue', async () => {
+	test('user should take the chat', async () => {
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
 		await agent.poHomeChannel.sidenav.openQueuedOmnichannelChat(newVisitor.name);
 		await expect(agent.poHomeChannel.content.btnTakeChat).toBeVisible();
 		await agent.poHomeChannel.content.btnTakeChat.click();
 
 		await agent.poHomeChannel.sidenav.openChat(newVisitor.name);
 		await expect(agent.poHomeChannel.content.btnTakeChat).not.toBeVisible();
 		await expect(agent.poHomeChannel.content.inputMessage).toBeVisible();
+
+		await poLiveChat.closeChat();
+	});
+
+	test('user should not receive a new chat when offline status and livechat disabled when agent idle', async ({ api }) => {
+		await api.post('/settings/Livechat_enabled_when_agent_idle', { value: false }).then((res) => expect(res.status()).toBe(200));
+		await agent.poHomeChannel.sidenav.switchStatus('offline');
+
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
+		await sleep(2000);
+
+		await expect(agent.poHomeChannel.sidenav.getQueuedChat(newVisitor.name)).toHaveCount(0);
 	});
 
-	test('expect "user1" to not able able to take chat from queue in-case its user status is offline', async () => {
-		// make "user-1" offline
+	test('user should not take the chat when offline status', async ({ api }) => {
+		await api.post('/settings/Livechat_enabled_when_agent_idle', { value: true }).then((res) => expect(res.status()).toBe(200));
+
+		await poLiveChat.openChatAndSendMessage(newVisitor, 'this_a_test_message_from_user');
+
 		await agent.poHomeChannel.sidenav.switchStatus('offline');
 
 		await agent.poHomeChannel.sidenav.openQueuedOmnichannelChat(newVisitor.name);
+
 		await expect(agent.poHomeChannel.content.btnTakeChat).toBeVisible();
+
 		await agent.poHomeChannel.content.btnTakeChat.click();
+		await expect(agent.page.locator('role=alert')).toBeVisible();
```

Comment:
to broad selector it could be any alert in the page, it can introduce new flaky test in the future if we have 2+ in the page